README.md

```diff
@@ -86,14 +86,14 @@ let string = import! std.string
 // Since imports in gluon returns regular values we can load specific parts of a module using pattern matches.
 // This match in particular brings in the equality operator for the `Char` type (this is required in
 // the current version of gluon but will be fixed in the future).
-let { eq = { (==) } } = import! std.char
+let char = import! std.char
 
-let { (<>) } = prelude.make_Semigroup string.semigroup
+let { (<>) } = import! std.prelude
```

Comment:
Why do we also not need to import the implicit instances as well?

Reply:
So currently I just search all reachable bindings marked as implicit and try to match them against the implicit type. This obviously extremely slow and doesn't allow implicit bindings to not be imported. That is, unless you are really careful to not make them reachable you will get a lot of (maybe) unintended implicit binding included. (That is really annoying since being careful when bringing in things in scope with `let int = import! std.int` actually has the opposite effect on implicits as all implicit bindings in the `int` record are made available for implicit resolution, whereas `let { num } = import! std.int` only brings in the `Num` instance).

Open for ideas on how to best represent importing `implicit` instances right now.

Reply:
Was thinking you could do have an implicit pattern thingy, eg:

```
let { [num] } = import! std.int
```
or:
```
let { num } = import! std.int
let [num] = num
```

Reply:
Just an idea! Not sure how well it would work with the function parameter syntax we were thinking of:

```
let { [num] } = import! std.int

let double : forall a. [Num a] -> a -> a = 
    \[num] x -> x + x

let double2 = double 2
```

We might also want to reserve `[pattern]` for pattern mathing on lists.... dilemmas!

Reply:
The disadvantage of the above would be you couldn't do:
```
let { .. } = import! std.int
```
... and have all the implicits come into scope. 🤔

Reply:
> let { [num] } = import! std.int

That is perhaps the most straightforward and explicit solution, but I wonder if it still ends up being to much work. All languages with type classes (that I can think of) tend to not need each instance to be imported. Maybe it isn't that bad though and I guess we could still have a syntax for "bring all implicit bindings into scope"

Reply:
> We might also want to reserve [pattern] for pattern matching on lists.... dilemmas!

Or arrays which currently get the expression syntax for `[expr]`. Still a bit torn on whether lists or arrays deserve this syntax. Lists are much more flexible than arrays with immutability, but on the other hand they should also be used sparingly...

Reply:
Scala requires you to import implicits. As does Agda, Coq, and Lean. If you have a globally coherent (and hence, non-modular) typeclass system like Haskell, Purescript, or Rust, then you don't need to.

Reply:
Yeah, we do need a way to limit what is considered in implicit resolution. I just meant that individually importing every single binding would be tedious so we need a way to "import every implicit binding in x"